scripts/src/checkprcontent/checkpr.py

```diff
@@ -207,6 +208,20 @@ def ensure_only_chart_is_modified(api_url, repository, branch):
             gitutils.add_output("pr-content-error-message", msg)
             sys.exit(1)
 
+        # Red Hat charts must carry the Red Hat prefix.
+        if organization == "redhat":
+            if not chart.startswith("redhat-"):
+                msg = f"[ERROR] Charts provided by Red Hat must have their name begin with the redhat- prefix. I.e. redhat-{chart}"
+                gitutils.add_output("pr-content-error-message", msg)
+                sys.exit(1)
+
+        # Non Red Hat charts must not carry the Red Hat prefix.
+        if organization != "redhat" and chart.startswith("redhat-"):
+            msg = f"[ERROR] The redhat- prefix is reserved for charts provided by Red Hat. Your chart: {chart}"
+            gitutils.add_output("pr-content-error-message", msg)
+            sys.exit(1)
```

Comment:
We are also enforcing charts under community/redhat to have this prefix, correct ?Seeing the conditions under which we run the .github/workflows/owners-redhat.yml workflow:
```
on:
  pull_request:
    paths:
    - charts/redhat/**/OWNERS
    - charts/community/redhat/**/OWNERS
```

If yes, this is going to fail here.

Reply:
@mgoerens we are only enforcing Red hat provided charts to have the `redhat-` prefix, while the `<vendor>-` prefix is being dropped for community and partner charts.

The reason was we didn't want Red Hat "squatting" on chart names for a community project or a partner's product (the latter case is much less likely).

Reply:
This is one of the earliest modifications I made, so please cross-check me intently! 

These code changes in checkpr check the **organization**, but not the **category**. That is, a chart that resides in `charts/redhat/redhat/myapp` has a category of redhat, an organization of redhat, and a chartname of myapp. 

A chart residing in `charts/community/redhat/anotherapp` has a category of community, an organization of redhat, and a chartname of anotherapp. 

The highlighted code snippet checks for **organization** values that are not redhat, and prevents them from using the prefix. Equally the workflow you mentioned (and earlier in checkpr) double checks to make sure submitted charts comply with the requirement to have the redhat prefix.

With that in mind, I believe the checkpr code is valid as written - but as mentioned, it's one of the earlier code snippets I wrote in the PoC, so please triple check me.

Reply:
> These code changes in checkpr check the organization, but not the category

I overlooked that. Thanks, that answers it then.